js/interface.js

```diff
@@ -1526,147 +1541,64 @@ var DynamicLists = (function() {
-
-        $(obj).html('');
-        $(obj).append('<option value="none">-- Select a data field</option>');
-
-        dataSourceColumns.forEach(function(value) {
-          options.push('<option value="' + value + '">' + value + '</option>');
-        });
-        $(obj).append(options.join(''));
-
-        if (oldValue && oldValue.length) {
-          $(obj).val(oldValue);
-        }
-      });
-
-      var emailOldValue = $('#select_user_email_data').val();
-      var options = [];
-
-      $('#select_user_email_data').html('');
-      $('#select_user_email_data').append('<option value="none">-- Select a data field</option>');
-      dataSourceColumns.forEach(function(value) {
-        options.push('<option value="' + value + '">' + value + '</option>');
-      });
-      $('#select_user_email_data').append(options.join(''));
-
-      if (emailOldValue && emailOldValue.length) {
-        $('#select_user_email_data').val(emailOldValue);
-      }
-
-      // Summary link field
-      var linkFieldValue = $('#select_field_link').val();
-      var linkOptions = [];
-
-      $('#select_field_link').html('');
-      linkOptions.push('<option value="none">-- Select a data field</option>');
-      linkOptions.push('<option disabled>------</option>');
-      dataSourceColumns.forEach(function(value) {
-        linkOptions.push('<option value="' + value + '">' + value + '</option>');
-      });
-      $('#select_field_link').append(options.join(''));
-
-      if (linkFieldValue && linkFieldValue.length) {
-        $('#select_field_link').val(linkFieldValue);
-      }
-
-      // Summary fields
-      $('[name="summary_select_field"]').each(function() {
-        var oldValue = $(this).val();
-        var defaultOptions = [
-          '<option value="none">-- Select a data field</option>',
-          '<option disabled>------</option>',
-          '<option value="empty">None</option>',
-          '<option value="custom">Custom</option>',
-          '<option disabled>------</option>'
-        ];
-        var options = _.concat(defaultOptions, _.map(dataSourceColumns, function(value) {
-          return '<option value="' + value + '">' + value + '</option>';
-        }));
-
-        $(this).html(options.join(''));
-
-        if (oldValue && oldValue.length) {
-          $(this).val(oldValue);
-        }
+      var options = dataSourceColumns.map(function(value) {
+        return '<option value="' + value + '">' + value + '</option>';
       });
+      var selectors = [
+        '#select_user_email_data',
+        '#select_chat_email',
+        '#select_field_link', // Summary link field
+        '#select_poll_data', // Pool data field
+        '#select_survey_data', // Survey data field
+        '#select_questions_data', // Questions data field
+        '[data-field="field"]',
+        '[name="summary_select_field"]', // Summary fields
+        '[name="detail_select_field"]' // Detail fields
+      ];
+
+      _.forEach(selectors, function(selector) {
```

Comment:
I'd rather do it this way to reduce the number of times jQuery has to query the DOM.

```js
$(selectors.join(',')).each(function() {
```

Then there's no need for `_.forEach(selectors, fn)` either.